Added documentation for goal preemption on behavior trees

gimait opened the pull request:
Signed-off-by: Aitor Miguel Blanco <aitormibl@gmail.com>

* Added new condition node to plugins index (navigation2 #[1712](https://github.com/ros-planning/navigation2/pull/1712)).
* Updated migration guide.
* Added new section for recommended tree structures in concepts, under Behavior Trees.

Other changes:
* Updated links in plugins index.
* Small typos and blank spaces at end of lines.

SteveMacenski commented on concepts/index.rst:
```diff
@@ -73,17 +73,44 @@ Additionally, modeling choices like to shoot at the goal from the left, right, o
+  :width: 50%
+  :align: center
+
+This tree structure allows having an asychronous node running (in this case the RecoveryActions node) while checking a condition (i.e., the navigation goal has been updated).
```
```suggestion
This tree structure allows having an asychronous node running (in this case the RecoveryActions node) while checking a condition (i.e., the navigation goal has been updated).
This allows for preemption of long-running tasks when a new goal comes in.
```

SteveMacenski commented on concepts/index.rst:
```diff
@@ -73,17 +73,44 @@ Additionally, modeling choices like to shoot at the goal from the left, right, o
 
+.. _useful-tree-structures:
+
+Useful Tree Structures
```
I don't think this is the place for this section. There's another ticket for adding documentation about BTs/designers guides/etc. This is just a quick introduction to the nomenclature and concepts.

Please remove this section. Maybe add this text to that ticket to add to those docs (or start on those separate docs, if you're interested :smile: ). This PR is just about the migration guide. Without this section, we can merge this PR

gimait replied:
I'll revert all changes in this file and take a look at that ticket. If nobody is working on it maybe I'll start with it.

SteveMacenski replied:
That would be great! I've asked someone else to also help out, but there's plenty to go around (and he hasn't agreed yet). I look at this like an entire subsection of the website at a high level like "tutorials" that contains several pages.

gimait commented on migration/Eloquent.rst:
```diff
@@ -33,14 +33,16 @@ DWB Contains new parameters as an update relative to the ROS1 updates, `see here
+See ``nav2_behavior_tree`` for a full listing, or :ref:`plugins` for the current list of behavior tree plugins and their descriptions.
 These plugins are set as default in the ``nav2_bt_navigator`` but may be overridden by the ``bt_plugins`` parameter to include your specific plugins.
 
 Original GitHub tickets:
```
yes, of course

gimait commented:
I reverted the changes in concepts index (except from the typo on BT Navigator, and the whitespaces) and added the goalupdated condition pr to the migration guide.
